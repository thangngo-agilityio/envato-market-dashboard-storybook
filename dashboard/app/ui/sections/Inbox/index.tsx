'use client';

import { useEffect, useMemo, useState } from 'react';
import {
  Text,
  Flex,
  theme,
  useColorModeValue,
  Grid,
  GridItem,
  useBreakpointValue,
  useToast,
} from '@chakra-ui/react';

// Firebase
import { convertTimeMessage, customToast, db } from '@/lib/utils';
import {
  DocumentData,
  DocumentReference,
  DocumentSnapshot,
  doc,
  getDoc,
  onSnapshot,
  setDoc,
} from 'firebase/firestore';

// Components
import {
  ChatMember,
  Conversation,
  EditIcon,
  FallbackImage,
} from '@/ui/components';

// Hooks
import { useGetUserDetails, useSubscribeToChat } from '@/lib/hooks';

// Store
import { authStore } from '@/lib/stores';
import { ERROR_MESSAGES, FIREBASE_CHAT, IMAGES, STATUS } from '@/lib/constants';

// Interfaces
import { TMessages } from '@/lib/interfaces';
import { useRouter, useSearchParams } from 'next/navigation';

const ChatMemberList = () => {
  const colorFill = useColorModeValue(
    theme.colors.gray[800],
    theme.colors.white,
  );
  const isMobile = useBreakpointValue({ base: true, lg: false });
  const router = useRouter();
  const searchParams = useSearchParams();
  const [chats, setChats] = useState<DocumentData | undefined>({});
  const [messages, setMessages] = useState<TMessages[]>([]);
  const [userInfo, setUserInfo] = useState({
    roomChatId: '',
    nameUser: '',
    adminUid: '',
    avatar: '',
    openRoom: false,
  });

  const { user: superAdmin } = authStore((state) => state);
  const uidUser = searchParams?.get('id') as string;
  const { filterDataUser } = useGetUserDetails(superAdmin?.id as string);
  const userChat = filterDataUser?.find((user) => user.uid === uidUser);
  const toast = useToast();

  const handleGetMessage = async (
    chatDocSnap: DocumentSnapshot<DocumentData, DocumentData>,
    chatDocRef: DocumentReference<DocumentData, DocumentData>,
    chatData: DocumentData | undefined,
    combinedId: string,
    nameUser: string,
    adminUid: string,
    avatar: string,
  ) => {
    !chatDocSnap.exists()
      ? await setDoc(chatDocRef, { messages: [] })
      : setMessages(chatData?.messages);

    setUserInfo({
      roomChatId: combinedId,
      nameUser: nameUser,
      adminUid: adminUid,
      avatar: avatar,
      openRoom: true,
    });
  };

  const handleSelectMember = async (user: {
    uid: string;
    avatarUrl: string;
    displayName: string;
  }) => {
    const id = user?.uid;
    router.push(`/inbox?id=${id}`);

    try {
      const combinedId = superAdmin?.uid + user.uid;
      const chatDocRef = doc(db, FIREBASE_CHAT.CHATS, combinedId);
      const chatDocSnap = await getDoc(chatDocRef);
      const chatData = chatDocSnap.data();

      await handleGetMessage(
        chatDocSnap,
        chatDocRef,
        chatData,
        combinedId,
        user.displayName,
        user.uid,
        user.avatarUrl,
      );
    } catch (error) {
      toast(
        customToast(
          ERROR_MESSAGES.SELECT_MEMBER_CHAT.title,
          ERROR_MESSAGES.SELECT_MEMBER_CHAT.description,
          STATUS.ERROR,
        ),
      );
    }
  };

  useEffect(() => {
    const getLastMessagesByUserId = async () => {
      try {
        const chatDocRef = doc(
          db,
          FIREBASE_CHAT.USER_CHATS,
          `${superAdmin?.uid}`,
        );
        const unsub = onSnapshot(chatDocRef, (doc) => {
          setChats(doc.data());
        });

        return () => {
          unsub();
        };
      } catch (error) {
        toast(
          customToast(
            ERROR_MESSAGES.LAST_MESSAGES_FAIL.title,
            ERROR_MESSAGES.LAST_MESSAGES_FAIL.description,
            STATUS.ERROR,
          ),
        );
      }
    };

    superAdmin?.uid && getLastMessagesByUserId();
  }, [toast, superAdmin?.uid]);

  const dataChats = useMemo(
    () => chats && Object.entries(chats)?.sort((a, b) => b[1].date - a[1].date),
    [chats],
  );

  useEffect(() => {
    const getRoomChat = async () => {
      if (uidUser) {
        const roomChatId = superAdmin?.uid + uidUser;
        const userInfo = chats && chats[roomChatId]?.userInfo;
        const chatDocRef = doc(db, FIREBASE_CHAT.CHATS, roomChatId);
        const chatDocSnap = await getDoc(chatDocRef);
        const chatData = chatDocSnap.data();

        await handleGetMessage(
          chatDocSnap,
          chatDocRef,
          chatData,
          roomChatId,
          userInfo
            ? userInfo.displayName
            : `${userChat?.firstName} ${userChat?.lastName}`,
          uidUser,
          userInfo ? userInfo?.avatarUrl : (userChat?.avatarURL as string),
        );
      }
    };

    superAdmin?.uid && getRoomChat();
  }, [
    chats,
    uidUser,
    superAdmin?.uid,
    userChat?.firstName,
    userChat?.lastName,
    userChat?.avatarURL,
  ]);

  useSubscribeToChat(userInfo.roomChatId, setMessages);

  return (
    <Grid
      templateColumns="repeat(12, minmax(0, 1fr))"
      borderTop="1px solid"
      borderColor="border.tertiary"
    >
      {isMobile ? (
        <GridItem
          colSpan={12}
          mb={4}
          padding={2}
          bg="background.body.septenary"
        >
          <Flex justify="flex-start" overflowX="scroll">
            {dataChats &&
              dataChats.map((chat) => (
                <ChatMember
                  key={chat[0]}
                  avatar={chat[1].userInfo?.avatarUrl}
                  onClick={() => handleSelectMember(chat[1].userInfo)}
                />
              ))}
          </Flex>
        </GridItem>
      ) : (
        <GridItem
          colSpan={4}
          bg="background.body.septenary"
          pt={6}
          pr={7}
          pl={12}
          pb={10}
          borderRight="1px solid"
          borderColor="border.tertiary"
          height="calc(100vh - 103px)"
          overflowY="scroll"
        >
          <Flex justify="space-between" align="center">
            <Text
              as="h3"
              color="text.primary"
              fontWeight="semibold"
              fontSize="20px"
            >
              Messages
              <Text
                as="span"
                color="text.primary"
                fontWeight="semibold"
                fontSize="20px"
              >
                ({chats ? Object.values(chats).length : 0})
              </Text>
            </Text>
            <EditIcon color={colorFill} />
          </Flex>
          <Flex direction="column" gap={6} py={3.5}>
            {dataChats &&
              dataChats.map((chat) => (
                <ChatMember
                  key={chat[0]}
                  avatar={chat[1].userInfo?.avatarUrl}
                  name={chat[1].userInfo?.displayName}
                  onClick={() => handleSelectMember(chat[1].userInfo)}
                  icon={
                    <FallbackImage
                      boxSize={4}
                      src={IMAGES.ATTACH.url}
                      alt={IMAGES.ATTACH.alt}
                    />
                  }
                  localeTime={convertTimeMessage(chat[1].date)}
                  lastMessages={chat[1]?.lastMessage?.text}
                />
              ))}
          </Flex>
        </GridItem>
      )}
      {userInfo.openRoom && (
        <GridItem colSpan={isMobile ? 12 : 8}>
          <Conversation
            nameUser={userInfo.nameUser}
            avatarUser={userInfo.avatar}
            messages={messages}
            adminUid={userInfo.adminUid}
          />
        </GridItem>
      )}
    </Grid>
  );
};

export default ChatMemberList;
