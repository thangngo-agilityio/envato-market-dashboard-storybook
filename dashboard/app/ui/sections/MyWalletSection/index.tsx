'use client';

import dynamic from 'next/dynamic';
import { InView } from 'react-intersection-observer';

//Components
import { Box, Flex } from '@chakra-ui/react';

// Lazy loading components
const TransactionTable = dynamic(
  () => import('@/ui/components/TransactionTable'),
);
const Efficiency = dynamic(() => import('@/ui/components/Efficiency'));
const OverallBalance = dynamic(() => import('@/ui/components/OverallBalance'));

const MyWalletsSection = () => (
  <InView>
    {({ inView, ref }) => (
      <Flex direction="column" gap={6} ref={ref}>
        <Flex
          flex={1}
          gap={6}
          direction={{ base: 'column', xl: 'row' }}
          boxSizing="border-box"
          w="100%"
        >
          <Box w={{ '3xl': '65%' }} flex={2}>
            {inView && <OverallBalance />}
          </Box>
          <Box w={{ '3xl': '35%' }} flex={1}>
            {inView && <Efficiency />}
          </Box>
        </Flex>
        <Box>
          <Box
            as="section"
            bgColor="background.component.primary"
            borderRadius={8}
            px={6}
            py={5}
          >
            {inView && <TransactionTable />}
          </Box>
        </Box>
      </Flex>
    )}
  </InView>
);

export default MyWalletsSection;
