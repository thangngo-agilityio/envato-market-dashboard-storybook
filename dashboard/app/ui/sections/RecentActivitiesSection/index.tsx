'use client';
import { InView } from 'react-intersection-observer';
import { Box, Flex, Grid, GridItem } from '@chakra-ui/react';
import dynamic from 'next/dynamic';

// dynamic loading components
const RecentActivitiesTable = dynamic(
  () => import('@/ui/components/RecentActivitiesTable'),
);
const CardPayment = dynamic(() => import('@/ui/components/CardPayment'));
const BoxChat = dynamic(() => import('@/ui/components/BoxChat'));

const RecentActivities = () => (
  <Grid
    bg="background.body.primary"
    py={12}
    px={{ base: 6, xl: 12 }}
    templateColumns={{ base: 'repeat(1, 1fr)', '2xl': 'repeat(4, 1fr)' }}
    display={{ sm: 'block', md: 'grid' }}
    gap={{ base: 0, '2xl': 12 }}
  >
    <GridItem colSpan={3}>
      <Box
        as="section"
        bgColor="background.component.primary"
        borderRadius={8}
        px={6}
        py={5}
      >
        <RecentActivitiesTable />
      </Box>
    </GridItem>
    <InView>
      {({ inView, ref }) => (
        <GridItem mt={{ base: 6, '2xl': 0 }} ref={ref}>
          <Flex direction={{ base: 'column', lg: 'row', xl: 'column' }} gap={6}>
            {inView && (
              <>
                <CardPayment /> <BoxChat />
              </>
            )}
          </Flex>
        </GridItem>
      )}
    </InView>
  </Grid>
);

export default RecentActivities;
