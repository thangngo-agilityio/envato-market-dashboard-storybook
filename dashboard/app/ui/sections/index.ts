export { default as UsersSection } from './UsersSection';
export { default as SettingsSection } from './SettingsSection';
export { default as SupportsSection } from './SupportSection';
export { default as ForgotPasswordSection } from './ForgotPassword';
export { default as AccountSection } from './AccountSection';
export { default as DashBoardSection } from './DashBoardSection';
export { default as MyWalletSection } from './MyWalletSection';
export { default as TransactionSection } from './TransactionSection';
export { default as HistorySection } from './HistorySection';
export { default as ProductsSection } from './ProductsSection';
export { default as RecentActivitiesSection } from './RecentActivitiesSection';
