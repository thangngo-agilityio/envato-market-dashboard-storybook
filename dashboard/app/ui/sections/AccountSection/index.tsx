'use client';

import { ReactNode, memo, useMemo } from 'react';
import { usePathname } from 'next/navigation';
import { Box, Flex } from '@chakra-ui/react';

// Components
import { Benefit, Divider, Logo, SwitchTheme } from '@/ui/components';
import Heading from '@/ui/layouts/AuthHeader';

// Layouts
import { AuthFooter } from '@/ui/layouts';

// Constants
import { ROUTES, TITLES } from '@/lib/constants';

type TAccountProps = {
  children?: ReactNode;
};

const AccountSection = ({ children }: TAccountProps): JSX.Element => {
  const pathname = usePathname();

  const pathForgotPassword = pathname === `/${ROUTES.FORGOT_PASSWORD}`;
  const pathLogin = pathname === `/${ROUTES.LOGIN}`;

  const title: string = useMemo(
    (): string =>
      pathForgotPassword
        ? TITLES.FORGOT_PASSWORD
        : pathLogin
          ? TITLES.SIGN_IN
          : TITLES.SIGN_UP,
    [pathForgotPassword, pathLogin],
  );

  return (
    <Flex width="100%" minH="100vh">
      <Box
        as="section"
        p="40px 0 48px"
        flex={1}
        w={{
          base: '100%',
          md: 'unset',
        }}
        bg="background.body.secondary"
      >
        <Flex justifyContent="space-between" px={12}>
          <Logo />
          <SwitchTheme />
        </Flex>
        <Box
          w={{
            base: '100%',
            sm: 425,
            md: 460,
          }}
          margin="auto"
          pt={24}
          pb={16}
          px={5}
          sx={{
            boxSizing: {
              base: 'border-box',
              md: 'unset',
            },
          }}
        >
          <Heading title={title} pathName={pathname} />
          <Divider content={!pathForgotPassword ? TITLES.AUTH_DiVIDER : ''} />
          {children}
          {pathForgotPassword && <AuthFooter />}
        </Box>
      </Box>
      <Benefit pathName={pathname} />
    </Flex>
  );
};

const Accounts = memo(AccountSection);

export default Accounts;
