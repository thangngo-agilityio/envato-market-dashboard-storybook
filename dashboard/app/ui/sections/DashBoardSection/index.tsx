'use client';

import lazy from 'next/dynamic';

import { InView } from 'react-intersection-observer';

// Components
import { Box, Grid, GridItem, Stack } from '@chakra-ui/react';
import { TotalStatisticList } from '@/ui/components';

// Lazy load components
const CardPayment = lazy(() => import('@/ui/components/CardPayment'));
const BoxChat = lazy(() => import('@/ui/components/BoxChat'));
// const TotalStatisticList = lazy(
//   () => import('@/ui/components/TotalStatisticList'),
// );
const RevenueFlow = lazy(() => import('@/ui/components/RevenueFlow'));
const Efficiency = lazy(() => import('@/ui/components/Efficiency'));
const TransactionTable = lazy(() => import('@/ui/components/TransactionTable'));

const DashBoardSection = () => (
  <Grid
    display={{ sm: 'block', md: 'grid' }}
    bg="background.body.primary"
    p={{ base: 6, xl: 12 }}
    templateColumns={{ base: 'repeat(1, 1fr)', '5xl': 'repeat(4, 1fr)' }}
    gap={0}
  >
    <GridItem colSpan={3}>
      <TotalStatisticList />

      <InView>
        {({ inView, ref }) => (
          <Grid
            templateColumns={{ base: 'repeat(1, 1fr)', lg: 'repeat(3, 1fr)' }}
            mt={6}
            gap={6}
            ref={ref}
            minH={250}
          >
            <GridItem colSpan={{ base: 3, xl: 2 }}>
              {inView && <RevenueFlow />}
            </GridItem>

            <GridItem ref={ref} display={{ base: 'none', xl: 'block' }}>
              {inView && <Efficiency />}
            </GridItem>
          </Grid>
        )}
      </InView>

      <InView>
        {({ inView, ref }) => (
          <Box
            ref={ref}
            mt={6}
            as="section"
            bgColor="background.component.primary"
            borderRadius={8}
            px={6}
            py={5}
            minH={500}
          >
            {' '}
            {inView && <TransactionTable />}
          </Box>
        )}
      </InView>
    </GridItem>

    <InView>
      {({ inView, ref }) => (
        <GridItem
          mt={{ base: 6, '5xl': 0 }}
          ml={{ '5xl': 12 }}
          ref={ref}
          minH={500}
        >
          {inView && (
            <Stack
              direction={{ base: 'column', lg: 'row', '2xl': 'column' }}
              spacing={{ base: 6, lg: 0 }}
            >
              <Box w="full">
                <CardPayment />
              </Box>

              <Box
                w="full"
                mt={{ base: 6, md: 0, '3xl': 6 }}
                ml={{ lg: 6, '2xl': 0 }}
              >
                <BoxChat />
              </Box>
            </Stack>
          )}
        </GridItem>
      )}
    </InView>
  </Grid>
);

export default DashBoardSection;
