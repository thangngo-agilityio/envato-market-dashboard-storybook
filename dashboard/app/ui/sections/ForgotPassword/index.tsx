'use client';

import { memo, useCallback } from 'react';
import Link from 'next/link';
import { Button, HStack, Text, VStack, Box, useToast } from '@chakra-ui/react';
import { Controller, SubmitHandler } from 'react-hook-form';

// Hooks
import { useForm } from '@/lib/hooks';

// Constants
import {
  ROUTES,
  AUTH_SCHEMA,
  SUCCESS_MESSAGES,
  STATUS,
  ERROR_MESSAGES,
} from '@/lib/constants';

// Components
import { InputField } from '@/ui/components';

// Firebase

import { sendPasswordResetEmail } from 'firebase/auth';

// Utils
import { auth, customToast } from '@/lib/utils';

type TForgotPasswordForm = {
  email: string;
};

const ForgotPasswordSection = (): JSX.Element => {
  // Control form
  const {
    control,
    formState: {
      errors: { root },
      isValid,
      isSubmitting,
    },
    reset,
    handleSubmit,
  } = useForm<TForgotPasswordForm>({
    defaultValues: {
      email: '',
    },
    mode: 'onBlur',
  });

  const toast = useToast();
  // TODO: Update later
  const handleSubmitForm: SubmitHandler<TForgotPasswordForm> = useCallback(
    async (data) => {
      try {
        await sendPasswordResetEmail(auth, data.email);
        toast(
          customToast(
            SUCCESS_MESSAGES.FORGOT_PASSWORD.title,
            SUCCESS_MESSAGES.FORGOT_PASSWORD.description,
            STATUS.SUCCESS,
          ),
        );
      } catch {
        toast(
          customToast(
            ERROR_MESSAGES.FORGOT_PASSWORD.title,
            ERROR_MESSAGES.FORGOT_PASSWORD.description,
            STATUS.ERROR,
          ),
        );
      } finally {
        reset();
      }
    },
    [reset, toast],
  );

  return (
    <VStack
      id="forgot-password-form"
      as="form"
      gap={6}
      onSubmit={handleSubmit(handleSubmitForm)}
    >
      <Controller
        rules={AUTH_SCHEMA.EMAIL}
        control={control}
        name="email"
        render={({ field, fieldState: { error } }) => (
          <InputField
            variant="authentication"
            placeholder="Enter your email"
            isError={!!error?.message}
            errorMessages={error?.message}
            isDisabled={isSubmitting}
            {...field}
          />
        )}
      />

      {/* Helpers */}
      <HStack justifyContent="end" w="100%">
        <Text
          as={Link}
          href={`/${ROUTES.LOGIN}`}
          aria-label="forgot password"
          color="text.currencyColor"
          fontWeight="semibold"
          textTransform="capitalize"
          textDecoration="underline"
        >
          back to sign in
        </Text>
      </HStack>

      {/* Show API error */}
      <Box w="full">
        <Text color="red" textAlign="center" py={2} h={10}>
          {root?.message}
        </Text>
        <Button
          aria-label="Recover Password"
          textTransform="capitalize"
          isDisabled={!isValid || isSubmitting}
          form="forgot-password-form"
          type="submit"
        >
          Recover Password
        </Button>
      </Box>
    </VStack>
  );
};

const ForgotPassword = memo(ForgotPasswordSection);

export default ForgotPassword;
