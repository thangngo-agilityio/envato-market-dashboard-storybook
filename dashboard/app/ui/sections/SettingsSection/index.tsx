'use client';

import { ReactElement, memo, useCallback, useState } from 'react';
import { Grid, GridItem } from '@chakra-ui/react';

// Constants
import { OPTION_SETTING } from '@/lib/constants';

// Components
import { Indicator, ItemSideBarSetting } from '@/ui/components';

// HOCs
import dynamic from 'next/dynamic';

// Lazy loading components
const AvatarSetting = dynamic(() =>
  import('@/ui/components/Icons/AvatarSetting').then((module) => ({
    default: module.AvatarSetting,
  })),
);

const Faq = dynamic(() =>
  import('@/ui/components/Icons/Faq').then((module) => ({
    default: module.Faq,
  })),
);

const Security = dynamic(() =>
  import('@/ui/components/Icons/Security').then((module) => ({
    default: module.Security,
  })),
);

const TermCondition = dynamic(() =>
  import('@/ui/components/Icons/TermCondition').then((module) => ({
    default: module.TermCondition,
  })),
);

const TermAndCondition = dynamic(
  () => import('@/ui/components/Setting/TermAndCondition'),
  {
    loading: () => <Indicator isOpen={true} />,
  },
);

const UserForm = dynamic(() => import('@/ui/components/Setting/Personal'), {
  loading: () => <Indicator isOpen={true} />,
});
const FaqPage = dynamic(() => import('@/ui/components/Setting/Faq'), {
  loading: () => <Indicator isOpen={true} />,
});
const SecurityPage = dynamic(() => import('@/ui/components/Setting/Security'), {
  loading: () => <Indicator isOpen={true} />,
});

const SettingsSection = () => {
  const [activeItemId, setActiveItemId] = useState<string>(
    OPTION_SETTING.USER_FORM,
  );

  const pages: Record<string, ReactElement> = {
    [OPTION_SETTING.USER_FORM]: <UserForm />,
    [OPTION_SETTING.FAQ_PAGE]: <FaqPage />,
    [OPTION_SETTING.SECURITY_PAGE]: <SecurityPage />,
    [OPTION_SETTING.TERM_AND_CONDITION]: <TermAndCondition />,
  };

  const handleItemClick = useCallback((id: string) => setActiveItemId(id), []);

  const renderSettingOptions = useCallback(() => {
    const ITEMS = [
      {
        id: OPTION_SETTING.USER_FORM,
        title: 'Personal Informations',
        content: 'Fill in your personal information',
        icon: <AvatarSetting />,
      },
      {
        id: OPTION_SETTING.FAQ_PAGE,
        title: 'FAQ',
        content: 'Frequently asked questions',
        icon: <Faq />,
      },
      {
        id: OPTION_SETTING.SECURITY_PAGE,
        title: 'Security',
        content: 'Change your password',
        icon: <Security />,
      },
      {
        id: OPTION_SETTING.TERM_AND_CONDITION,
        title: 'Term and Conditions',
        content: 'Term and Conditions of use and privacy policy',
        icon: <TermCondition />,
      },
    ];

    return ITEMS.map(({ id, icon, title, content }) => (
      <ItemSideBarSetting
        key={id}
        id={id}
        activeItemId={activeItemId}
        onClick={handleItemClick}
        title={title}
        content={content}
      >
        {icon}
      </ItemSideBarSetting>
    ));
  }, [activeItemId, handleItemClick]);

  return (
    <Grid
      bg="background.body.primary"
      borderRadius="lg"
      gridTemplateColumns={{
        base: 'repeat(1,minmax(0,1fr))',
        xl: 'repeat(12,minmax(0,1fr))',
      }}
      px={{ base: 6, md: 12 }}
      py={12}
    >
      <GridItem px={4} py={6} colSpan={3} bg="background.body.quaternary">
        {renderSettingOptions()}
      </GridItem>

      <GridItem colSpan={9} px={10} py={8} bg="background.body.quaternary">
        {pages[activeItemId]}
      </GridItem>
    </Grid>
  );
};

const Settings = memo(SettingsSection);

export default Settings;
