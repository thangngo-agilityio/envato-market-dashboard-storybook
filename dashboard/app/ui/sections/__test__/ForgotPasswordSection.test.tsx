// Sections
import { ForgotPasswordSection } from '@/ui/sections';
import userEvent from '@testing-library/user-event';
import * as firebase from 'firebase/auth';

const { render, waitFor } = testLibReactUtils;

const sendPasswordResetEmailMock = jest.fn();

describe('ForgotPasswordSection render', () => {
  beforeEach(() => {
    jest
      .spyOn(firebase, 'sendPasswordResetEmail')
      .mockImplementation(sendPasswordResetEmailMock);
  });

  test('Should render match with snapshot.', () => {
    const { container } = render(<ForgotPasswordSection />);

    expect(container).toMatchSnapshot();
  });

  test('Handle submit recover password', async () => {
    const { getByPlaceholderText, getByRole } = render(
      <ForgotPasswordSection />,
    );

    const input = getByPlaceholderText('Enter your email');
    const button = getByRole('button', { name: /recover password/i });

    await userEvent.type(input, 'canhvo0603@gmail.com');
    await userEvent.click(button);

    waitFor(() => {
      expect(sendPasswordResetEmailMock).toHaveBeenCalled();
    });
  });
});
