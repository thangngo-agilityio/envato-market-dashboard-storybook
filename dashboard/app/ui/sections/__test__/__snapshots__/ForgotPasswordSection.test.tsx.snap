// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordSection render Should render match with snapshot. 1`] = `
<div>
  <form
    class="chakra-stack css-1urha0v"
    id="forgot-password-form"
  >
    <div
      class="chakra-form-control css-1kxonj9"
      role="group"
    >
      <div
        class="chakra-input__group css-1091msq"
        data-group="true"
      >
        <input
          class="chakra-input css-offuse"
          id="field-:r0:"
          name="email"
          placeholder="Enter your email"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="chakra-stack css-mdisnt"
    >
      <a
        aria-label="forgot password"
        class="chakra-text css-1f9o43a"
        href="/login"
      >
        back to sign in
      </a>
    </div>
    <div
      class="css-1ki54i"
    >
      <p
        class="chakra-text css-rj1t1b"
      />
      <button
        aria-label="Recover Password"
        class="chakra-button css-1nt97yu"
        disabled=""
        form="forgot-password-form"
        type="submit"
      >
        Recover Password
      </button>
    </div>
  </form>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
